Sources/ThresholdKey/Modules/TssSecurityQuestionModule.swift

```diff
@@ -0,0 +1,242 @@
+public final class TssSecurityQuestionModule {
+    
+    /// set security question
+    /// - Parameters:
+    ///   - threshold_key: The threshold key to act on.
+    ///   - question: The security question.
+    ///   - answer: The answer for the security question.
+    ///   - factorKey: Factor key that registred to security question
+    ///   - tag: tss tag
+    ///
+    /// - Returns: ``
+    ///
+    /// - Throws: `RuntimeError`, indicates invalid parameters was used or invalid threshold key of failed to set security question
+    public static func set_security_question( threshold : ThresholdKey, question: String, answer: String, factorKey :String, tag: String ) async throws {
+        //
+        let domainKey = TssSecurityQuestion + ":" + tag
+        
+        var isSet = false
+        do {
+            let question = try TssSecurityQuestionModule.get_question(threshold: threshold, tag: tag)
+            if question.count > 0 {
+                isSet = true
+            }
+        } catch {}
+        
+        if isSet {throw "Trying to set Security Question again"}
+        
+        
+        guard let hash = answer.data(using: .utf8)?.sha3(.keccak256) else {
+            throw "Invalid answer format"
+        }
+        
+        let hashKey = PrivateKey(hex: hash.toHexString())
+        let hashPub = try hashKey.toPublic();
+        guard let encryptedData = try encrypt(key: hashPub, msg: factorKey).data(using: .utf8) else {
+            throw "encryption error"
+        }
+        
+        let encryptedMsg = try JSONDecoder().decode(EncryptedMessage.self, from: encryptedData)
+
+        let factorPub = try PrivateKey(hex: factorKey).toPublic(format: .EllipticCompress)
+        
+        let shareIndex = "3"
+        
+        let data = TssSecurityQuestionData( shareIndex: shareIndex, factorPublicKey: factorPub, associatedFactor: encryptedMsg, question: question )
+        
+        
+        let jsonData = try JSONEncoder().encode(data)
+        guard let jsonStr = String(data: jsonData, encoding: .utf8) else {
+            throw "Invalid security question data"
+        }
+        try threshold.set_general_store_domain(key: domainKey, data: jsonStr )
+        
+        try await threshold.sync_metadata();
+    }
+    
+    
+    /// change security question to new question and answer
+    /// - Parameters:
+    ///   - threshold_key: The threshold key to act on.
+    ///   - newQuestion: The new security question .
+    ///   - newAnswer: The new answer for the security question.
+    ///   - answer: current answer
+    ///   - tag: tss tag
+    ///
+    /// - Returns: ``
+    ///
+    /// - Throws: `RuntimeError`, indicates invalid parameters was used or invalid threshold key or fail to change security question
+    public static func change_security_question( threshold : ThresholdKey, newQuestion: String, newAnswer: String, answer: String, tag: String) async throws {
```

Comment:
why tag is required, cant we just read it from threshold key, we are not taking tag as input in web.